src/com/epam/javatrainings/genericannotations/util/validators/AdulthoodValidator.java

```diff
@@ -0,0 +1,40 @@
+package com.epam.javatrainings.genericannotations.util.validators;
+
+import com.epam.javatrainings.genericannotations.annotations.Adulthood;
+
+import java.lang.reflect.Field;
+import java.time.LocalDate;
+import java.util.List;
+
+public class AdulthoodValidator<T> {
+
+    private AdulthoodValidator() {
+    }
+
+    public static <T> boolean validateAdulthood(T t, List<String> errorMessages) {
+        boolean isValid = true; //It means that is no annotation - no restriction
+        try {
+            Field[] fields = t.getClass().getDeclaredFields();
+            Field birthDate = null;
+            for (Field field : fields) {
```

Comment:
Can be iterated the object fields once in GlobalValidator.validate method, if on the field declared the corresponding annotation then validate it with the corresponding validator.